hermes/bin/src/runtime_extensions/hermes/ipfs/state/api.rs

```diff
@@ -0,0 +1,128 @@
+//! Hermes IPFS State API
+use super::{is_valid_dht_content, is_valid_pubsub_content, HERMES_IPFS_STATE};
+use crate::{
+    app::HermesAppName,
+    runtime_extensions::bindings::hermes::ipfs::api::{
+        DhtKey, DhtValue, Errno, IpfsContent, IpfsFile, IpfsPath, MessageData, MessageId, PeerId,
+        PubsubTopic,
+    },
+};
+
+/// Add File to IPFS
+pub(crate) fn hermes_ipfs_add_file(
+    app_name: &HermesAppName, contents: IpfsFile,
+) -> Result<IpfsPath, Errno> {
+    tracing::debug!(app_name = %app_name, "adding IPFS file");
+    let ipfs_path = HERMES_IPFS_STATE.file_add(contents)?;
+    tracing::debug!(app_name = %app_name, path = %ipfs_path, "added IPFS file");
+    HERMES_IPFS_STATE
+        .apps
+        .added_file(app_name.clone(), ipfs_path.clone());
+    Ok(ipfs_path)
+}
+
+/// Validate IPFS Content from DHT or `PubSub`
+pub(crate) fn hermes_ipfs_content_validate(
+    app_name: &HermesAppName, content: &IpfsContent,
+) -> bool {
+    match content {
+        IpfsContent::Dht((k, v)) => {
+            // TODO(@saibatizoku): Implement types and validation
+            let key_str = format!("{k:x?}");
+            let is_valid = is_valid_dht_content(k, v);
+            tracing::debug!(app_name = %app_name, dht_key = %key_str, is_valid = %is_valid, "DHT value validation");
+            is_valid
+        },
+        IpfsContent::Pubsub((topic, message)) => {
+            // TODO(@saibatizoku): Implement types and validation
```

Comment:
I think these TODOs can be removed right? Since there's TODOs inside the is_valid_* functions